src/runtime/heap/mem_debug.c

```diff
@@ -203,23 +222,23 @@ static void mem_debug_backed_dealloc_unmap(backed_heap h, void *v, u64 p, bytes
 
 static u64 mem_debug_backed_alloc(heap h, bytes b)
 {
-    return u64_from_pointer(mem_debug_backed_alloc_map((backed_heap)h, b, 0));
+    return u64_from_pointer(alloc_map((backed_heap)h, b, 0));
```

Comment:
We should be able to use mem_debug_backed_alloc_map_nohdr() here, which should work regardless of whether the nohdr option has been specified on heap creation.

Reply:
I'm not following you I think--the alloc_map/dealloc_unmap function used should depend on the nohdr flag at creation, as a normal alloc/dealloc should still use the header checks if that's how it was set up. Always using the nohdr versions would skip the boundary and header checks.

Reply:
You are right, somehow I was under the impression that invoking alloc_map() here would call the non-debug version (thereby skipping all checks) of the backed heap allocation function. But now I see that's clearly not the case, since we still operate on a memdebug heap.